### PR TITLE
Fix(indev): Don't leave edit mode if there is only one object in the group

### LINKED ISSUE #1995: When there is only one object in the group, it still leave edit mode.
**Describe the bug**

When I use the encoder as input and there is only one object in the group, long press LV_KEY_ENTER will still leave edit mode.

**Expected behavior**

Don't leave edit mode if there is only one object in the group.

**Solution**

1. Change [src/lv_core/lv_indev.c line 672](https://github.com/lvgl/lvgl/blob/master/src/lv_core/lv_indev.c#L672) from `if(_lv_ll_is_empty(&g->obj_ll) == false) {` to `if(_lv_ll_is_empty(&g->obj_ll) == false && _lv_ll_get_len(&g->obj_ll) > 1 ) {`
```
/*Don't leave edit mode if there is only one object (nowhere to navigate)*/
if(_lv_ll_is_empty(&g->obj_ll) == false && _lv_ll_get_len(&g->obj_ll) > 1 ) {
    lv_group_set_editing(g, lv_group_get_editing(g) ? false : true); /*Toggle edit mode on long press*/
}
```
2. Fix the bug that the long press release event isn't sent to the object in this case. Change [src/lv_core/lv_indev.c line 738](https://github.com/lvgl/lvgl/blob/master/src/lv_core/lv_indev.c#L738) from `if(!i->proc.long_pr_sent || _lv_ll_is_empty(&g->obj_ll)) {` to `if(!i->proc.long_pr_sent || _lv_ll_is_empty(&g->obj_ll) || _lv_ll_get_len(&g->obj_ll) <= 1) {`
```
/*Ignore long pressed enter release because it comes from mode switch*/
if(!i->proc.long_pr_sent || _lv_ll_is_empty(&g->obj_ll) || _lv_ll_get_len(&g->obj_ll) <= 1) {
    indev_obj_act->signal_cb(indev_obj_act, LV_SIGNAL_RELEASED, NULL);
    if(indev_reset_check(&i->proc)) return;

    lv_event_send(indev_obj_act, LV_EVENT_SHORT_CLICKED, NULL);
    if(indev_reset_check(&i->proc)) return;

    lv_event_send(indev_obj_act, LV_EVENT_CLICKED, NULL);
    if(indev_reset_check(&i->proc)) return;

    lv_event_send(indev_obj_act, LV_EVENT_RELEASED, NULL);
    if(indev_reset_check(&i->proc)) return;

    lv_group_send_data(g, LV_KEY_ENTER);
}
```

### PR DESCRIPTION
**Pull request by udoudou**
### Description of the feature or fix

Fixes [#1995 ](https://github.com/lvgl/lvgl/issues/1995)
This bug seems to exist in V6 and later branch versions.

### Checkpoints
- [x] Follow the [styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [ ] Update CHANGELOG.md
- [ ] Update the documentation

### CODE REVIEW

File: src/lv_core/lv_indev.c [hunk 1]
```diff
@@ -669,7 +669,7 @@ static void indev_encoder_proc(lv_indev_t * i, lv_indev_data_t * data)
                 /*On enter long press toggle edit mode.*/
                 if(editable) {
                     /*Don't leave edit mode if there is only one object (nowhere to navigate)*/
-                    if(_lv_ll_is_empty(&g->obj_ll) == false) {
+                    if(_lv_ll_is_empty(&g->obj_ll) == false && _lv_ll_get_len(&g->obj_ll) > 1 ) {
```
Comment (kisvegabor) [inline src/lv_core/lv_indev.c]: I think it should be simply:
```suggestion
                    if (_lv_ll_get_len(&g->obj_ll) > 1 ) {
```

File: src/lv_core/lv_indev.c [hunk 2]
```diff
@@ -735,7 +735,7 @@ static void indev_encoder_proc(lv_indev_t * i, lv_indev_data_t * data)
             /*An object is being edited and the button is released. */
             else if(g->editing) {
                 /*Ignore long pressed enter release because it comes from mode switch*/
-                if(!i->proc.long_pr_sent || _lv_ll_is_empty(&g->obj_ll)) {
+                if(!i->proc.long_pr_sent || _lv_ll_is_empty(&g->obj_ll) || _lv_ll_get_len(&g->obj_ll) <= 1) {
```
Comment (kisvegabor) [inline src/lv_core/lv_indev.c]: Same here:
```suggestion
                if(!i->proc.long_pr_sent || _lv_ll_get_len(&g->obj_ll) <= 1) {
```

**Review by kisvegabor — COMMENTED** [review]
Thank you!

Could you also add a line to [CHANGELOG](https://github.com/lvgl/lvgl/blob/master/CHANGELOG.md)?

**Comment by udoudou** [comment]
I updated the CHANGELOG in [#2011 ](https://github.com/lvgl/lvgl/pull/2011)

**Comment by kisvegabor** [comment]
Thank you!

**Comment by kisvegabor** [comment]
I've merged it into `master`.

**Pull request merged**